src/rpc/rpc_base_conn.js

```diff
@@ -129,12 +144,37 @@ class RpcBaseConnection extends EventEmitter {
         return this._state === STATE_CLOSED;
     }
 
+    _encode_message(msg) {
```

Comment:
jsdoc:
```suggestion
    /**
     * @param {RpcMessage} msg
     * @returns {Buffer[]}
     */
    _encode_message(msg) {
```

Reply:
This will break as this does not return buffers in all cases, especially for the case of fcall

Reply:
OK so you can use `@returns {Buffer[] | object}`

Reply:
Or set a `/** @template T */` on the RpcBaseConn class and that will be the encoded message type, so that `RpcFcallConn extends RpcBaseConn<object>` but the others `extends RpcBaseConn<Buffer[]>`.

Reply:
done